src/components/DataTable.ts

```diff
@@ -47,131 +50,154 @@ import './datatable.scss';
     DataTableFooter,
     Visibility
   ],
-  host: {
-    '[class.fixed-header]': 'options.headerHeight !== "auto"',
-    '[class.fixed-row]': 'options.rowHeight !== "auto"',
-    '[class.scroll-vertical]': 'options.scrollbarV',
-    '[class.scroll-horz]': 'options.scrollbarH',
-    '[class.selectable]': 'options.selectable',
-    '[class.checkboxable]': 'options.checkboxable'
-  },
-  providers: [ StateService ]
+  providers: [StateService]
 })
-export class DataTable {
+export class DataTable implements OnInit, DoCheck, AfterViewInit {
 
   @Input() options: TableOptions;
-  @Input() rows: Array<any>;
-  @Input() selected: Array<any>;
+  @Input() rows: any[];
+  @Input() selected: any[];
 
   @Output() onPageChange: EventEmitter<any> = new EventEmitter();
   @Output() onRowsUpdate: EventEmitter<any> = new EventEmitter();
   @Output() onRowClick: EventEmitter<any> = new EventEmitter();
   @Output() onSelectionChange: EventEmitter<any> = new EventEmitter();
   @Output() onColumnChange: EventEmitter<any> = new EventEmitter();
 
-  @ContentChildren(DataTableColumn) columns;
+  @ContentChildren(DataTableColumn) columns: QueryList<DataTableColumn>;
 
   private element: HTMLElement;
-  private rowDiffer: any;
-  private colDiffer: any;
+  private rowDiffer: IterableDiffer;
+  private colDiffer: IterableDiffer;
+
+  @HostBinding('class.datatable') isDatatable: boolean = true;
 
   constructor(element: ElementRef, private state: StateService, differs: KeyValueDiffers) {
     this.element = element.nativeElement;
-    this.element.classList.add('datatable');
-    
     this.rowDiffer = differs.find({}).create(null);
     this.colDiffer = differs.find({}).create(null);
   }
 
-  ngOnInit() {
-    let { options, rows, selected } = this;
+  ngOnInit(): void {
+    const {options, rows, selected} = this;
 
     this.state
       .setOptions(options)
       .setRows(rows)
       .setSelected(selected);
   }
 
-  ngAfterViewInit() {
+  ngAfterViewInit(): void {
     this.adjustColumns();
 
-    setTimeout(() => {
-      for(let col of this.columns.toArray()) {
-        this.options.columns.push(new TableColumn(col));
-      }
-    });
+    if (this.columns.length) {
```

Comment:
Be careful with `setTimeout` because it triggers change detection for the hole app